lnrpc/devrpc/dev_server.go

```diff
@@ -0,0 +1,342 @@
+		return nil, fmt.Errorf("unable to decode output index: %v", err)
+	}
+
+	txid, err := chainhash.NewHashFromStr(split[0])
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse hex string: %v", err)
+	}
+
+	return &wire.OutPoint{
+		Hash:  *txid,
+		Index: uint32(index),
+	}, nil
+}
+
+func parsePubKey(pubKeyStr string) ([33]byte, error) {
+	var pubKey [33]byte
+	pubKeyBytes, err := hex.DecodeString(pubKeyStr)
+	if err != nil || len(pubKeyBytes) != 33 {
+		return pubKey, fmt.Errorf("invalid pubkey: %v", pubKeyStr)
+	}
+
+	copy(pubKey[:], pubKeyBytes)
+	return pubKey, nil
+}
+
+// ImportGraph imports a graph dump (without auth proofs).
+//
+// NOTE: Part of the DevServer interface.
+func (s *Server) ImportGraph(ctx context.Context,
+	graph *lnrpc.ChannelGraph) (*ImportGraphResponse, error) {
+
+	// Obtain the pointer to the global singleton channel graph.
+	graphDB := s.cfg.GraphDB
+
+	var err error
+	for _, rpcNode := range graph.Nodes {
+		node := &channeldb.LightningNode{
+			HaveNodeAnnouncement: true,
+			LastUpdate: time.Unix(
+				int64(rpcNode.LastUpdate), 0,
+			),
+			Alias: rpcNode.Alias,
+		}
+
+		node.PubKeyBytes, err = parsePubKey(rpcNode.PubKey)
+		if err != nil {
+			return nil, err
+		}
+
+		featureBits := make([]lnwire.FeatureBit, 0, len(rpcNode.Features))
+		featureNames := make(map[lnwire.FeatureBit]string)
+
+		for featureBit, feature := range rpcNode.Features {
+			featureBits = append(
+				featureBits, lnwire.FeatureBit(featureBit),
+			)
+
+			featureNames[lnwire.FeatureBit(featureBit)] = feature.Name
+		}
+
+		featureVector := lnwire.NewRawFeatureVector(featureBits...)
+		node.Features = lnwire.NewFeatureVector(
+			featureVector, featureNames,
+		)
+
+		node.Color, err = lncfg.ParseHexColor(rpcNode.Color)
+		if err != nil {
+			return nil, err
+		}
+
+		if err := graphDB.AddLightningNode(node); err != nil {
+			return nil, fmt.Errorf("unable to add node %v: %v",
+				rpcNode.PubKey, err)
+		}
+
+		log.Debugf("Imported node: %v", rpcNode.PubKey)
+	}
+
+	for _, rpcEdge := range graph.Edges {
+		rpcEdge := rpcEdge
+
+		edge := &channeldb.ChannelEdgeInfo{
+			ChannelID: rpcEdge.ChannelId,
+			ChainHash: *s.cfg.ActiveNetParams.GenesisHash,
+			Capacity:  btcutil.Amount(rpcEdge.Capacity),
+		}
+
+		edge.NodeKey1Bytes, err = parsePubKey(rpcEdge.Node1Pub)
+		if err != nil {
+			return nil, err
+		}
+
+		edge.NodeKey2Bytes, err = parsePubKey(rpcEdge.Node2Pub)
+		if err != nil {
+			return nil, err
+		}
+
+		channelPoint, err := parseOutPoint(rpcEdge.ChanPoint)
+		if err != nil {
+			return nil, err
+		}
+		edge.ChannelPoint = *channelPoint
+
+		if err := graphDB.AddChannelEdge(edge); err != nil {
+			return nil, fmt.Errorf("unable to add edge %v: %v",
+				rpcEdge.ChanPoint, err)
+		}
+
+		makePolicy := func(rpcPolicy *lnrpc.RoutingPolicy) *channeldb.ChannelEdgePolicy {
+			policy := &channeldb.ChannelEdgePolicy{
+				ChannelID: rpcEdge.ChannelId,
+				LastUpdate: time.Unix(
+					int64(rpcPolicy.LastUpdate), 0,
```

Comment:
Same time question here. Also we we need to validate any other fields, or are they checked later on when we do the import in `UpdateEdgePolicy`?

Reply:
Similarly what's a valid timestamp for a routing policy update?

Reply:
Surely this can't be zero?